staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -298,7 +300,12 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 		}
 
 		if err := r.watchHandler(w, &resourceVersion, resyncerrc, stopCh); err != nil {
-			if err != errorStopRequested {
+			switch err {
+			case errorStopRequested:
+				// no extra log message
+			case etcdrpc.ErrCompacted:
```

Comment:
this layer doesn't currently have a dependency on or knowledge of etcd... can we propagate this in a way that doesn't make us deep compare to specific etcd errors?

Reply:
actually... an API client isn't going to get raw etcd errors... it will get an API error

Reply:
lemme rework that

Reply:
_sigh_ ok fixed that